instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/internal/SpanKeyProvider.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.internal;
+
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
+import java.util.stream.Stream;
+
+/**
+ * Returns {@link SpanKey}s associated with the {@link AttributesExtractor} that implements this
+ * interface.
+ *
+ * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
+ * at any time.
+ */
+public interface SpanKeyProvider {
+
+  Stream<SpanKey> internalGetSpanKeys();
+}
```

Comment:
I like the flip.

two thoughts/questions:

* maybe ok to assume at most one SpanKey and return `@Nullable SpanKey`?
* is it internal just out of caution? seems like a nice hook for anyone defining their own semantic conventions

cc @lmolkova

Reply:
> * is it internal just out of caution? seems like a nice hook for anyone defining their own semantic conventions

It is a nice hook - but since `SpanKey` is internal I wanted to be doubly sure and made this method internal too.

Reply:
makes sense, we can think about making it public later

wdyt about assuming at most one SpanKey instead of returning `Stream`?

Reply:
Ah, I somehow missed that in you previous comment - yes, that makes sense; I don't think we'll ever need to associate >1 span keys with a single attributes extractor.